security/pkg/cmd/flag.go

```diff
@@ -33,3 +34,52 @@ func InitializeFlags(rootCmd *cobra.Command) {
 	_ = flagSet.Parse([]string{})
 	flag.CommandLine = flagSet
 }
+
```

Comment:
If we have to do custom flag processing - most likely something is wrong. 

We want to eventually merge the 2 agents, and to allow the code to be reused as a
SDS ( secret discovery service envoy API) - we should avoid deps on custom flag processing.

In general for agent it would be highly valuable to use env variables where possible and
minimize dependencies.

Reply:
I think the requirement is legitimate. The node agent requires different sets of arguments to bootstrap for different platforms (onprem, aws, gcp, etc.). And we want to have the processing of the platform-specific flags separate from other flags.

For configuration of propagating the credentials to the backend, including through SDS, we are using the common flag set in init(). How would the custom flag processing affect this?
 
I think using env variables to replace arguments could be done separately in the future.

Reply:
First, this is not custom flag processing; it's only filtering unknown flags and the flag processing is still done by pflag library. I have also added tests for the function.

Second, we can use env variables approach. However I think there may be some backward compatibility issues for people who is already using our node agent for onprem. @myidpt may have more context on this.

Reply:
Sorry I accidentally deleted my last comment.

For env variable approach, I don't think we need to worry too much about backward compatibility now because Istio is still in its early stage :)